src/box/memtx_tree.cc

```diff
@@ -932,6 +965,138 @@ memtx_tree_index_get_internal(struct index *base, const char *key,
 	return 0;
 }
 
+/**
+ * Get position of iterator - extracted cmp_def of last fetched tuple.
+ * If iterator is exhausted, it returns position of last tuple. If there were
+ * no tuples fetched, pos and size are not updated.
+ */
+template <bool USE_HINT>
+static int
+memtx_tree_index_iterator_position(struct index *base, struct iterator *it,
+				   const char **pos, uint32_t *size)
+{
+	assert(base != NULL);
+	assert(it != NULL);
+	assert(pos != NULL);
+	assert(size != NULL);
+
+	struct errinj *inj = errinj(ERRINJ_ITERATOR_POSITION_FAIL,
+				    ERRINJ_BOOL);
+	if (inj != NULL && inj->bparam == true) {
+		diag_set(ClientError, ER_INJECTION, "iterator_position fail");
+		return -1;
+	}
+
+	struct memtx_tree_index<USE_HINT> *index =
+		(struct memtx_tree_index<USE_HINT> *)base;
+	struct key_def *cmp_def = memtx_tree_cmp_def(&index->tree);
+	struct tree_iterator<USE_HINT> *tree_it =
+		get_tree_iterator<USE_HINT>(it);
+	struct tuple *tuple = tree_it->current.tuple;
+	if (tuple == NULL)
+		return 0;
+	*pos = tuple_extract_key(tuple, cmp_def, MULTIKEY_NONE, size);
+	return *pos != NULL ? 0 : -1;
+}
+
+/**
+ * Version of memtx_tree_index_iterator_position for multikey index.
+ */
+static int
+memtx_tree_index_iterator_position_multikey(struct index *base,
```

Comment:
Let's join it with `memtx_tree_index_iterator_position` using template parameter.

Reply:
done